app/client/components/auth/registerForm.jsx

```diff
@@ -23,224 +21,139 @@ const mapDispatchToProps = dispatch => {
-  } else if (props.invalid_register) {
-    error_message = props.invalid_register;
-  } else if (props.password_invalid) {
-    error_message = "Password too weak";
-  } else {
-    error_message = "";
+  if (props.register_status.success) {
+    return (
+      <div>
+        <h3> Register Success </h3>
+        <div> We've sent you an activation email</div>
+      </div>
+    );
   }
 
   return (
-    <div>
-      <div style={{ display: "block" }}>
-        <p
-          style={
-            props.state.invite
-              ? { margin: "1em 0.5em", textAlign: "center" }
-              : { display: "none" }
-          }
-        >
-          {props.state.username}
-        </p>
-
-        <p
-          style={
-            props.state.invite
-              ? { margin: "1em 0.5em", textAlign: "center", fontWeight: 600 }
-              : { display: "none" }
-          }
-        >
-          Please choose a password
-        </p>
-
+    <Form
+      form={form}
+      name="basic"
+      style={{ maxWidth: "300px" }}
+      onFinish={onFinish}
+      onFinishFailed={onFinishFailed}
+    >
+      <p
+        style={
+          props.email
+            ? { margin: "1em 0.5em", textAlign: "center", fontWeight: 600 }
+            : { display: "none" }
+        }
+      >
+        Please choose a password
+      </p>
+      <Form.Item
+        aria-label="Email"
+        name="email"
+        rules={[{ required: true, message: "Please enter your email!" }]}
+      >
         <Input
-          type="email"
-          onKeyUp={props.onUserFieldKeyPress}
           placeholder="Email"
-          disabled={props.state.invite ? "disabled" : ""}
-          style={props.state.invite ? { display: "none" } : null}
-          aria-label="Email"
+          type={"email"}
+          disabled={props.email ? "disabled" : ""}
         />
-
-        <ReactPasswordStrength
-          minLength={8}
-          type="password"
-          changeCallback={data => props.onPasswordFieldKeyPress(data)}
-          inputProps={{ placeholder: "Password" }}
-          className={"default-input"}
+      </Form.Item>
+      <Form.Item
+        name="password"
+        rules={[
+          ({ getFieldValue }) => ({
+            validator(_, value) {
+              if (!value) {
+                return Promise.reject("Please enter your password!");
+              }
+              if (value && value.length <= 8) {
+                return Promise.reject(
+                  "Password must be at least 8 characters long"
+                );
+              }
+              if (value !== getFieldValue("retypePassword")) {
+                return Promise.reject("Passwords do not match");
+              }
+              return Promise.resolve();
+            }
+          })
+        ]}
+        aria-label="password"
+        dependencies={["retypePassword"]}
+      >
+        <PasswordInput
+          inputProps={{
+            placeholder: "Password",
+            minLength: "8",
```

Comment:
Not sure why, but it's rejecting an 8 character password (error message goes away on character 9)
![image](https://user-images.githubusercontent.com/2199643/128568946-4d92e944-f789-40ad-bea3-d90f13b2eccc.png)
![image](https://user-images.githubusercontent.com/2199643/128568979-10031d63-e3da-4a5a-b741-371adf545b0f.png)

Reply:
This was the issue: https://github.com/teamsempo/SempoBlockchain/pull/739#discussion_r684902854